crates/rethnet_rpc_server/src/lib.rs

```diff
@@ -201,7 +341,7 @@ async fn handle_set_storage_at(
     position: U256,
     value: U256,
 ) -> ResponseData<()> {
-    match (*state)
+    match (*state.rethnet)
```

Comment:
Does `state.rethnet.write` work?

Reply:
i think you're looking at an old/outdated revision? as explained [here](https://github.com/NomicFoundation/hardhat/pull/3998#discussion_r1236001000), i made these changes across two different commits (one in the "block spec handling" commit, and another specific to the remaining instances that didn't have anything to do with block spec handling).